include/llvm/CodeGen/PagerandoBinning.h

```diff
@@ -44,11 +44,18 @@ class PagerandoBinning : public ModulePass {
   static constexpr unsigned BinSize = 4096; // one page
   static constexpr unsigned MinFnSize = 2;  // 'bx lr' on ARM thumb
 
-  std::multimap<unsigned, unsigned> Bins;   // <free space  ->  bin numbers>
-  unsigned BinCount = 1;
-
-  unsigned assignToBin(const MachineFunction &MF);
   unsigned estimateFunctionSize(const MachineFunction &MF);
+
+public:
+  class Algorithm {
```

Comment:
What is the purpose of this class? It seems fine, just could use a comment here explaining what it is for.